subprojects/core/src/main/java/org/gradle/plugin/repository/portal/CustomPluginRepositorySpec.java

```diff
@@ -0,0 +1,49 @@
+/*
+ * Copyright 2016 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.plugin.repository.portal;
+
+import org.gradle.api.Incubating;
+import org.gradle.api.artifacts.dsl.RepositoryHandler;
+import org.gradle.api.artifacts.repositories.ArtifactRepository;
+
+/**
+ * When implemented, a user will be able to provide custom plugin resolution logic.
+ *
+ * TODO: Better name
+ */
+@Incubating
+public interface CustomPluginRepositorySpec {
+
+    /**
+     * Used to add any repository into the resolution engine.
+     *
+     * @param repositoryHandler a RepositoryHandler
+     */
+    ArtifactRepository registerRepositories(RepositoryHandler repositoryHandler);
```

Comment:
Because we want to prepend instead of append repositories. The global plugin repositories should have precedence over the ones defined in the buildscript block of a project. It's an unlikely edge case, users will rarely mix the two.